tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -2168,6 +2168,66 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  /** Translate litterals to there ScalaJS String representation. */
+  private def foldToStringForString_+(tree: Tree)(implicit pos : Position): Tree = tree match {
+    case FloatLiteral(value)   => StringLiteral(jsNumberToString(value.toDouble))
+    case DoubleLiteral(value)  => StringLiteral(jsNumberToString(value))
+    case LongLiteral(value)    => StringLiteral(value.toString)
+    case IntLiteral(value)     => StringLiteral(value.toString)
+    case BooleanLiteral(value) => StringLiteral(value.toString)
+    case Null()                => StringLiteral("null")
+    case Undefined()           => StringLiteral("undefined")
+
+    case _ => tree
+  }
+
+  /* Following the ECMAScript 6 specification */
+  private def jsNumberToString(value: Double): String = {
+    @inline def getJVMnks : (Int, Int, BigInt) = {
+      
+      val jvmString = value.toString
+      val n = math.floor(math.log10(value)).toInt + 1
+      
+      val sString = jvmString.filter(_ != '.').takeWhile(_.isDigit)
+
+      // The JVM does love its .0
+      val corrected = sString.stripSuffix("0")
+      
+      val s = BigInt(corrected)
+      val k = corrected.length
+
+      (n, k, s)
+    }
+
+    if (1.0.toString == "1") { 
+      // We are in a JS environment, so the host .toString() is the correct one.
+      value.toString 
+    } else value match {
```

Comment:
We don't use that style in this codebase. Put `value match {` in a nested block:
```scala
} else {
  value match {
    ...
  }
}
```